benches/bench.rs

```diff
@@ -0,0 +1,26 @@
+use ak09915_rs::{Ak09915, Mode, Register};
+use criterion::{criterion_group, criterion_main, Criterion};
+use linux_embedded_hal::I2cdev;
+
+fn new() -> Ak09915<I2cdev> {
+    let dev = I2cdev::new("/dev/i2c-1").unwrap();
+    Ak09915::new(dev)
+}
+
+fn ak09915(c: &mut Criterion) {
+    #[macro_export]
+    macro_rules! bench {
+    ($bench_fn:ident($($arg:tt)*)) => {
+        let mut sensor = new();
+        sensor.set_mode(Mode::Cont200Hz).unwrap();
+        c.bench_function(stringify!($bench_fn), |b| b.iter(|| sensor.$bench_fn($($arg)*)));
+    }}
+
+    c.bench_function("new", |b| b.iter(|| new()));
+
+    bench!(read_register(Register::ST1));
```

Comment:
We are calling the functions but not checking if there is an error.

Reply:
We return an error, if we fail fast.. it's not a valid benchmark test..
We should check if the return it's at least valid

Reply:
https://github.com/bheisler/criterion.rs/issues/506

Reply:
@patrickelectric Please check, it works and doesn't affect performance

Reply:
why just not `.expect("Error during benchmark");`?

Reply:
@RaulTrombin how are we checking making it valid if `read_register` can return an error ? How are we currently validating it ? @joaoantoniocardoso suggestion makes more sense.

Reply:
thks @joaoantoniocardoso 
changed to just expect, just retesting.

@patrickelectric it will panic and stop the whole benchmark on errors

Reply:
tested on pi